uberfire-extensions/uberfire-wires/uberfire-wires-core/uberfire-wires-core-grids/src/test/java/org/uberfire/ext/wires/core/grids/client/util/CellContextUtilitiesTest.java

```diff
@@ -257,6 +265,55 @@ public void testMakeRenderContextNoBlock() {
                      0.0);
     }
 
+    @Test
+    public void testEditWhenNoCellSelected() {
+        final GridColumn.HeaderMetaData headerMetaData = mock(GridColumn.HeaderMetaData.class);
+        final GridColumn<?> gridColumn = mockGridColumn(100.0, Arrays.asList(headerMetaData));
+        gridWidget.getModel().appendColumn(gridColumn);
+
+        CellContextUtilities.editSelectedCell(gridWidget);
+
+        verify(headerMetaData, never()).edit(any(GridBodyCellEditContext.class));
+        verify(gridWidget, never()).startEditingCell(anyInt(), anyInt());
+        verify(gridWidget, never()).startEditingCell(any(Point2D.class));
+    }
+
+    @Test
+    public void testEditWhenHeaderCellSelected() {
+        final GridColumn.HeaderMetaData headerMetaData = mock(GridColumn.HeaderMetaData.class);
+        final GridColumn<?> gridColumn = mockGridColumn(100.0, Arrays.asList(headerMetaData));
+
+        doReturn(ci).when(gridRendererHelper).getColumnInformation(50.0);
+        doReturn(gridColumn).when(ci).getColumn();
+        doReturn(0.0).when(ci).getOffsetX();
+        doReturn(0).when(ci).getUiColumnIndex();
+
+        gridWidget.getModel().appendColumn(gridColumn);
+        gridWidget.getModel().selectHeaderCell(0, 0);
+
+        CellContextUtilities.editSelectedCell(gridWidget);
+
+        verify(headerMetaData).edit(any(GridBodyCellEditContext.class));
+        verify(gridWidget, never()).startEditingCell(anyInt(), anyInt());
+        verify(gridWidget, never()).startEditingCell(any(Point2D.class));
+    }
+
+    @Test
+    public void testEditWhenDataCellSelected() {
+        final GridColumn.HeaderMetaData headerMetaData = mock(GridColumn.HeaderMetaData.class);
+        final GridColumn<?> gridColumn = mockGridColumn(100.0, Arrays.asList(headerMetaData));
+        final GridRow gridRow = new BaseGridRow();
+
+        gridWidget.getModel().appendColumn(gridColumn);
+        gridWidget.getModel().appendRow(gridRow);
+        gridWidget.getModel().selectCell(0, 0);
```

Comment:
I always try to avoid identical co-ordinates `(0, 0)` as you don't know that the verification of L#314 has not transposed the co-ordinates.. I personally prefer something like `(0, 1)`.. but a test is better than no test :-)

Reply:
@manstis agree, will try to improve